bundles/org.eclipse.ui.workbench/Eclipse UI/org/eclipse/ui/internal/handlers/TraversePageHandler.java

```diff
@@ -38,20 +40,90 @@ public class TraversePageHandler extends WidgetMethodHandler {
 	public final Object execute(final ExecutionEvent event) {
 		Control focusControl = Display.getCurrent().getFocusControl();
 		if (focusControl != null) {
+			boolean forward = "next".equals(methodName); //$NON-NLS-1$
 			int traversal = "next".equals(methodName) ? SWT.TRAVERSE_PAGE_NEXT : SWT.TRAVERSE_PAGE_PREVIOUS; //$NON-NLS-1$
 			Control control = focusControl;
 			do {
-				if (control.traverse(traversal))
+				if (control instanceof CTabFolder folder && isFinalItemInCTabFolder(folder, forward)
+						&& !areHiddenItems(folder)) {
+
+					loopToSecondToFirstItemInCTabFolder(folder, forward);
+					traversal = getTraversalDirection(!forward); // we are in the second-to-last item in the given
+					// direction. Now, use the Traverse-event to move back by one
+				}
+				if (control.traverse(traversal)) {
 					return null;
-				if (control instanceof Shell)
+				}
+				if (control instanceof Shell) {
 					return null;
+				}
 				control = control.getParent();
 			} while (control != null);
 		}
 
 		return null;
 	}
 
+
+	private int getTraversalDirection(boolean direction) {
+		return direction ? SWT.TRAVERSE_PAGE_NEXT : SWT.TRAVERSE_PAGE_PREVIOUS;
+	}
+
+	/**
+	 * Sets the current selection to the second-to-last item in the given direction.
+	 *
+	 * @param folder
+	 * @param forward
+	 */
+	private void loopToSecondToFirstItemInCTabFolder(CTabFolder folder, boolean forward) {
+		if (forward) {
+			folder.showItem(folder.getItem(0));
+			folder.setSelection(1);
+		} else {
+			int itemCount = folder.getItemCount();
+			folder.setSelection(itemCount - 2);
+		}
+	}
+
+	/**
+	 * {@return Returns whether the folder has currently selected the final item in
+	 * the given direction.}
+	 *
+	 * @param folder  the CTabFolder which we want to inspect
+	 * @param forward whether we want to traverse forwards of backwards
+	 */
+	private boolean isFinalItemInCTabFolder(CTabFolder folder, boolean forward) {
+		folder.update();
+
+		CTabItem currentFolder = folder.getSelection();
+		CTabItem lastFolder = null;
+		if (forward) {
+			int itemCount = folder.getItemCount();
+
+			lastFolder = folder.getItem(itemCount - 1);
+
+		} else {
+			lastFolder = folder.getItem(0);
+		}
+		return currentFolder.equals(lastFolder);
+	}
+
+	/**
+	 * checks if there are any hidden items
+	 *
+	 * @param folder
+	 * @return if there are any items which are not shown in the folder (tabs in the
+	 *         >> column)
```

Comment:
the ">> column" is called "the chevron"

Reply:
https://ux.stackexchange.com/questions/105135/difference-between-caret-and-chevron

Reply:
done